File: milestone1/tests/test1.py
def bubbleSort(array: list[int]) -> None:
  i: int = 0
  for i in range(len(array)):
    swapped: bool = False
    for j in range(0, len(array) - i - 1):
      if array[j] > array[j + 1]:
        temp: int = array[j]
        array[j] = array[j + 1]
        array[j + 1] = temp
        swapped = True
    if not swapped:
      break


def insertionSort(arr: list[int]) -> None:
  i: int = 0
  for i in range(1, len(arr)):
    key: int = arr[i] 
    j: int = i - 1
    while j >= 0 and key < arr[j]:
      arr[j + 1] = arr[j]
      j -= 1
    arr[j + 1] = key


def main():
  data1: list[int] = [-2, 45, 0, 11, -9]
  data2: list[int] = [-2, 45, 0, 11, -9]

  bubbleSort(data1)
  insertionSort(data2)

  print('Sorted array using bubble sort:')
  i: int = 0
  for i in range(len(data1)):
    print(data1[i])
  
  print('Sorted array using insertion sort:')
  for i in range(len(data2)):
    print(data2[i])


if __name__ == "__main__":
  main()

File: milestone3/x86-examples/test1.c
#include <stdio.h>

void bubbleSort(int array[], int n) {
    int i, j;
    for (i = 0; i < n; i++) {
        int swapped = 0;
        for (j = 0; j < n - i - 1; j++) {
            if (array[j] > array[j + 1]) {
                int temp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = temp;
                swapped = 1;
            }
        }
        if (!swapped)
            break;
    }
}

void insertionSort(int arr[], int n) {
    int i, j;
    for (i = 1; i < n; i++) {
        int key = arr[i];
        j = i - 1;
        while (j >= 0 && key < arr[j]) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = key;
    }
}

int main() {
    int data1[] = {-2, 45, 0, 11, -9};
    int data2[] = {-2, 45, 0, 11, -9};

    int n = sizeof(data1) / sizeof(data1[0]);

    bubbleSort(data1, n);
    insertionSort(data2, n);

    for (int i = 0; i < n; i++) {
        printf("%d\n", data1[i]);
    }

    printf("\n");
    for (int i = 0; i < n; i++) {
        printf("%d\n", data2[i]);
    }

    return 0;
}
